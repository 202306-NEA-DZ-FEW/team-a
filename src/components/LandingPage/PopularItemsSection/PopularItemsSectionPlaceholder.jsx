import Container from "@/components/container";

function PopularItemsSectionPlaceholder() {
  const fakeData = [1, 2, 3, 4];

  return (
    <Container className='flex flex-col justify-center my-10'>
      <div className='w-1/3 rounded-full h-8 self-center mb-14 bg-gray-300 animate-pulse' />
      <div className='flex justify-center items-center flex-wrap gap-4'>
        {fakeData.map((d) => (
          <div
            key={d}
            className='h-64 w-52 rounded-3xl bg-gray-300 animate-pulse'
          />
        ))}
      </div>
    </Container>
  );
}

export default PopularItemsSectionPlaceholder;
