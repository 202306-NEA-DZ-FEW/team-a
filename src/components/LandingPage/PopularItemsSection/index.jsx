import Container from "@/components/container";
import ProductCard from "@/components/ProductCard";
import { useTranslation } from "next-i18next";

function PopularItemsSection() {
  const { t } = useTranslation();
  const items = [
    {
      id: 1,
      title: "sweater",
      location: "djelfa",
      description: "lorem ipsum dolor sit amet, consectetur adipiscing...",
      tag: "swap",
      category: "clothes",
      imageUrl:
        "https://images.unsplash.com/photo-1560060141-7b9018741ced?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1479&q=80",
    },
    {
      id: 2,
      title: "sofa",
      location: "medea",
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing...",
      tag: "donation",
      category: "furniture",
      imageUrl:
        "https://images.unsplash.com/photo-1540574163026-643ea20ade25?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80",
    },
    {
      id: 3,
      title: "headsets",
      location: "sba",
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing...",
      tag: "request",
      category: "electronics",
      imageUrl:
        "https://images.unsplash.com/photo-1545127398-14699f92334b?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1335&q=80",
    },
    {
      id: 4,
      title: "hat",
      location: "algiers",
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing...",
      tag: "swap",
      category: "clothes",
      imageUrl:
        "https://images.unsplash.com/photo-1521369909029-2afed882baee?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1470&q=80",
    },
    {
      id: 5,
      title: "poncho",
      location: "batna",
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing...",
      tag: "donation",
      category: "clothes",
      imageUrl:
        "https://images.unsplash.com/photo-1434389677669-e08b4cac3105?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=1410&q=80",
    },
  ];
  return (
    <Container className='lg:min-h-screen my-20 flex flex-col justify-center'>
      <h1 className='text-3xl md:text-5xl font-bold text-center mb-14'>
        {t("landingPage:items")}
      </h1>
      <div className='flex justify-center items-center flex-wrap gap-4'>
        {items.map((item) => (
          <ProductCard key={item.id} {...item} />
        ))}
      </div>
    </Container>
  );
}
export default PopularItemsSection;
