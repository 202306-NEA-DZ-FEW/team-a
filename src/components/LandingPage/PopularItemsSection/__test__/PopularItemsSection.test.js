import renderer from "react-test-renderer";

import PopularItemsSection from "..";

it("renders correctly", () => {
  const mockItems = [
    {
      id: 1,
      title: "Hello",
      images: ["https://placehold.co/600x400"],
      createdAt: {},
      description: "Hello",
    },
  ];
  const tree = renderer
    .create(<PopularItemsSection items={mockItems} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
