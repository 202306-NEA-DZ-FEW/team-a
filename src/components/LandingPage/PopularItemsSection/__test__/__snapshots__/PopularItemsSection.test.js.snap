// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="max-w-6xl mx-auto px-8 xl:px-0 lg:min-h-screen my-20 flex flex-col justify-center"
>
  <h1
    className="text-3xl md:text-5xl font-bold text-center mb-14"
  >
    items
  </h1>
  <div
    className="flex justify-center items-center flex-wrap gap-4"
  >
    <a
      href="/items/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="card rounded-3xl group"
      >
        <figure
          className="relative h-64 w-52 rounded-3xl"
        >
          <img
            alt="Hello"
            className="group-hover:scale-110 saturate-100 group-hover:blur-[2px] hover:saturate-50 brightness-[0.73] object-cover w-full h-full duration-500"
            data-nimg="1"
            decoding="async"
            height={200}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
            src="/_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=3840&q=75"
            srcSet="/_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=3840&q=75 3840w"
            style={
              Object {
                "backgroundImage": "url(\\"data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 200 200'%3E%3Cfilter id='b' color-interpolation-filters='sRGB'%3E%3CfeGaussianBlur stdDeviation='20'/%3E%3CfeColorMatrix values='1 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 100 -1' result='s'/%3E%3CfeFlood x='0' y='0' width='100%25' height='100%25'/%3E%3CfeComposite operator='out' in='s'/%3E%3CfeComposite in2='SourceGraphic'/%3E%3CfeGaussianBlur stdDeviation='20'/%3E%3C/filter%3E%3Cimage width='100%25' height='100%25' x='0' y='0' preserveAspectRatio='none' style='filter: url(%23b);' href='public/images/blur.jpg'/%3E%3C/svg%3E\\")",
                "backgroundPosition": "50% 50%",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "cover",
                "color": "transparent",
              }
            }
            width={200}
          />
        </figure>
        <div
          className="absolute bottom-[5%] flex flex-col px-3 w-52"
        >
          <h3
            className="text-white text-lg font-bold capitalize"
          >
            Hello
          </h3>
          <span
            className="text-sm text-white pl-1 font-light mb-1"
          >
            undefined
          </span>
          <div
            className="flex items-center gap-1 p-0 w-full"
          >
            <span
              className="badge badge-md font-light flex gap-1 text-white border-opacity-30 bg-black bg-opacity-25"
            >
              undefi...
              <span>
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M13.02 19.93v2.02c2.01-.2 3.84-1 5.32-2.21l-1.42-1.43a7.941 7.941 0 01-3.9 1.62zM4.03 12c0-4.05 3.03-7.41 6.95-7.93V2.05C5.95 2.58 2.03 6.84 2.03 12c0 5.16 3.92 9.42 8.95 9.95v-2.02c-3.92-.52-6.95-3.88-6.95-7.93zM19.95 11h2.02c-.2-2.01-1-3.84-2.21-5.32l-1.43 1.43c.86 1.1 1.44 2.43 1.62 3.89zM18.34 4.26a9.981 9.981 0 00-5.32-2.21v2.02c1.46.18 2.79.76 3.9 1.62l1.42-1.43zM18.33 16.9l1.43 1.42A9.949 9.949 0 0021.97 13h-2.02a7.941 7.941 0 01-1.62 3.9z"
                  />
                  <path
                    d="M16 11.1C16 8.61 14.1 7 12 7s-4 1.61-4 4.1c0 1.66 1.33 3.63 4 5.9 2.67-2.27 4-4.24 4-5.9zm-4 .9a1.071 1.071 0 010-2.14A1.071 1.071 0 0112 12z"
                  />
                </svg>
              </span>
            </span>
            <span
              className="badge badge-md font-light text-white border-white border-opacity-30 bg-black bg-opacity-25"
            >
              undefined
            </span>
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;
